core/src/main/java/org/elasticsearch/monitor/jvm/JvmStats.java

```diff
@@ -79,14 +79,15 @@ public static JvmStats jvmStats() {
                 MemoryUsage usage = memoryPoolMXBean.getUsage();
                 MemoryUsage peakUsage = memoryPoolMXBean.getPeakUsage();
                 String name = GcNames.getByMemoryPoolName(memoryPoolMXBean.getName(), null);
-                if (name == null) { // if we can't resolve it, its not interesting.... (Per Gen, Code Cache)
-                    continue;
+                if (name == null) { // if we can't resolve it, its not GC related.... (Metaspace, Code Cache)
+                    name = memoryPoolMXBean.getName();
                 }
+                // MemoryUsage.getMax() can be -1, indicating the maximum is undefined (e.g. metaspace max)
                 pools.add(new MemoryPool(name,
-                        usage.getUsed() < 0 ? 0 : usage.getUsed(),
-                        usage.getMax() < 0 ? 0 : usage.getMax(),
-                        peakUsage.getUsed() < 0 ? 0 : peakUsage.getUsed(),
-                        peakUsage.getMax() < 0 ? 0 : peakUsage.getMax()
+                        usage.getUsed(),
+                        usage.getMax() == -1 ? Long.MAX_VALUE : usage.getMax(),
```

Comment:
We can. It does mean people have to be prepared to understand -1 means unbounded, not unsupported, but in general I am for less "interpretation". I would just change them to writeLong and not play around. The numbers will be big and trying to compress them is overkill.